common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -0,0 +1,85 @@
+package com.microsoft.identity.common.internal.providers.microsoft.microsoftsts;
+
+import com.microsoft.identity.common.Account;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectory;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryCloud;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.ClientInfo;
+import com.microsoft.identity.common.internal.providers.oauth2.AccessToken;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.IDToken;
+import com.microsoft.identity.common.internal.providers.oauth2.OAuth2Strategy;
+import com.microsoft.identity.common.internal.providers.oauth2.RefreshToken;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenResponse;
+
+import java.net.URL;
+
+public class MicrosoftStsOAuth2Strategy extends OAuth2Strategy {
+
+    private MicrosoftStsOAuth2Configuration mConfig;
+
+    public MicrosoftStsOAuth2Strategy(MicrosoftStsOAuth2Configuration config) {
+        super(config);
+        mConfig = config;
+    }
+
+    @Override
+    public String getIssuerCacheIdentifier(AuthorizationRequest request) {
+        if (request instanceof MicrosoftStsAuthorizationRequest) {
+            final URL authority = ((MicrosoftStsAuthorizationRequest) request).getAuthority();
+            // TODO I don't think this is right... This is probably not the correct authority cache to consult...
```

Comment:
Eventually remove this TODO, depending on if the instance metadata moves somewhere else or not....

Reply:
Are you using this for the refresh token or he access token or both?